### DISCUSSION TITLE
Batch Processing Feature

### DISCUSSION THREAD

**Discussion by Kraftwurm** [mifi/lossless-cut#1910] [Ideas]
Hi there

LosslessCut is a wonderful peace of polished software! I use it regularly and I am a long time supporter over at [OpenCollective](https://opencollective.com)!

My Request:

- Would it be feasible to have a "Batch Processing Feature" ?

I would love to have a feature where I can also just say "Add To Batch" instead of "Export", which will postpone the actual work for later. That way I could do all the "manual things" like setting the clipping points or preparing the "merge/concatenate" stuff in advance and when I'm done I could let it work overnight for example to do the actual processing, whatever it is.

For me this would be an extremely useful feature that would greatly enhance an already very useful software program!

Kind Regards,
Jimmy Gee

**Comment by mifi** [comment]
Hi and thanks for your support! this has been discussed here #89 - and it's not a trivial rewrite. see in particular https://github.com/mifi/lossless-cut/issues/89#issuecomment-1682437333 hopefully some day but it takes a lot of effort.

**Reply by Kraftwurm** [reply]
Hi Mikael!

Thanks for the reply and sorry for not having read through all the comments before!

**I have an IDEA!**

Yes, I totally see the problems and I have an idea that would solve all the problems that you mention [here](https://github.com/mifi/lossless-cut/issues/89#issuecomment-1682437333) kind of elegantly if implemented properly. (Which seems to be your magic trick anyway ;)

You already allow **multiple instances of the program** to run and there would be a seemingly "hacky" solution that - when implemented properly - could be an acceptable solution to the problem that solves everything elegantly:

Just allow some kind of "letting multiple instances run in parallel but (by communication between the running instances) **only allow ONE instance at a time to do the actual processing**.

That way, we could then just open a new instance for every new project, do all the manual preparations etc. and then just click on "PROCESS", which would either immediately start the processing OR (!) just be forced to wait until other, earlier instances are done processing.

There could evet be a flag in the configuration to "force serialization of processing/or not".

What do you think about it?

Kind Regards,
Jimmy

**Reply by mifi** [reply]
hmm, you mean each time you edit a new file in the batch list you open a new instanec of losslesscut? then say you have 200 files in the batch list, then you open 200 instances of losslesscut and when you're done you press "export" and each instance starts exporting one by one? I don't think most computers would have enough memory to run even close to that many instances unfortunately

**Comment by Kraftwurm** [comment]
I see.

Electron Apps are big, I think that's the problem, right?

Well, no luck with my idea this time. Thanks anyway!

--
Sent from Missive <https://missiveapp.com/?ref=signature> - Email, Chat &
Tasks for Productive Teams

**Reply by mifi** [reply]
Yes but i think any app can only run X instances on a normal computer. Try to open any video editor 10 times and it will probably grind the computer to a halt, so it’s not a very scalable solution
